impl/src/main/java/org/ehcache/internal/store/offheap/OffHeapStore.java

```diff
@@ -141,7 +138,7 @@ public OffHeapStore(final Configuration<K, V> config, Serializer<K> keySerialize
         public OffHeapValueHolder<V> apply(K mappedKey, OffHeapValueHolder<V> mappedValue) {
           long now = timeSource.getTimeMillis();
 
-          if (mappedValue == null || mappedValue.isExpired(now)) {
+          if (mappedValue == null || mappedValue.isExpired(now, OffHeapValueHolder.TIME_UNIT)) {
```

Comment:
Using the `OffHeapValueHolder.TIME_UNIT` here seems fishy without also converting now into it. Why don't you simply pass `TimeUnit.MILLISECONDS`?
This remark applies to all calls like this and also to equivalent ones in `OnHeapStore`.